src/compiler/mod.rs

```diff
@@ -797,6 +812,36 @@ lazy_static::lazy_static! {
             })),
         ));
 
+        builtins.push((
+            "indexOf".to_string(),
+            Some(TypeConstraint::Array),
+            Function::Builtin(Box::new(&|compiler, _, args| {
+                ensure!(args.len() == 2, ArgumentsCountSnafu { found: args.len(), expected: 2usize });
+
+                let old_root_scope = compiler.root_scope;
+                compiler.root_scope = &BUILTINS_SCOPE;
+                let result = array::find_index(compiler, &args[0], &args[1])?;
+                compiler.root_scope = old_root_scope;
+
+                Ok(result)
+            })),
+        ));
+
+        builtins.push((
+            "includes".to_string(),
+            Some(TypeConstraint::Array),
+            Function::Builtin(Box::new(&|compiler, _, args| {
+                ensure!(args.len() == 2, ArgumentsCountSnafu { found: args.len(), expected: 2usize });
+
+                let old_root_scope = compiler.root_scope;
+                compiler.root_scope = &BUILTINS_SCOPE;
+                let result = array::includes(compiler, &args[0], &args[1])?;
+                compiler.root_scope = old_root_scope;
```

Comment:
Perhaps we should update scope once?

Reply:
The BUILTINS_SCOPE contains the hidden/system/unsafe builtins and safe builtins. We need to reset it back to the user's old root scope that only has the safe builtins + root functions + collections.